core/synch.c

```diff
@@ -2247,6 +2248,8 @@ detach_on_permanent_stack(bool internal, bool do_cleanup)
 
     LOG(GLOBAL, LOG_ALL, 1, "Detach: Entering final cleanup and unload\n");
     SYSLOG_INTERNAL_INFO("Detaching from process, entering final cleanup");
+    if (drstats != NULL)
+        dr_get_stats(drstats);
```

Comment:
The impl of dr_get_stats is only there for CLIENT_INTERFACE, so even though this built it will have an unresolved import and will fail to run for LD_BIND_NOW.  Given the "promotion" of dr_stats_t, maybe move dr_get_stats() into core/utils.c?